stag-library-compiler/src/test/java/com/vimeo/stag/processor/CleanableJavaFileManager.java

```diff
@@ -0,0 +1,103 @@
+package com.vimeo.stag.processor;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.concurrent.LinkedBlockingQueue;
+
+import javax.tools.FileObject;
+import javax.tools.JavaFileManager;
+import javax.tools.JavaFileObject;
+
+/**
+ * {@link JavaFileManager} implementation which tracks created files and allows them to be
+ * systematically deleted.
+ */
+class CleanableJavaFileManager implements JavaFileManager {
```

Comment:
It is not.  I spun this up in response to test artifacts (i.e., generated source) being left in the module's root dir.  It actually still leaves empty directories around but git doesn't care since it doesn't do directories natively.